### DISCUSSION TITLE
Automated server side decline of too low offers and adding to transfer market again

### DISCUSSION THREAD

**Discussion by eLindros** [fabfischer/kickbase-plus#53] [Ideas]
Hi @fabfischer ,

I think about some server side logic to automatically decline too low offers and adding those players to the transfer market again. This should be checked and run every our. It also needs to be customizable. I published Kickbase-plus on Netlify where I could use scheduled functions (like cronjobs). 

Did you have any thoughts about that?

**Comment by fabfischer** [comment]
Hi @eLindros 

We could add such a feature for people who host this project on their own webserver/cloud-hosting, which provides nodejs-capabilities (or cloud functions in your case).

But I see several edge cases:

- We need to have an auto-login feature, which will login for the user. Therefore the user's credentials must be stored somewhere on the server, or in env variables.
- Sometimes I don't want to decline specific offers. For example personally I don't decline an offer which might be the last offer before market values changes are calculated by kickbase. And on match days, I don't decline an offer that may not be the best, but could save my ass. In short: I think, the auto-declining feature would need a rule set feature - as you suggested. But this may be tricky in the details.

But I am open to such a feature. Have you already started with it?

**Reply by eLindros** [reply]
> We need to have an auto-login feature, which will login for the user. Therefore the user's credentials must be stored somewhere on the server, or in env variables.

What about storing only the token on the server? Do you know how long the token is valid?

**Reply by fabfischer** [reply]
The token is valid for 2 weeks as far I know

**Comment by eLindros** [comment]
Hi @fabfischer,

I see all your points and have to think about them.

I started a little but then got sidetracked. But it annoys me so much that I have to check Kickbase every hour that I will definitely give it a try. I'll let you know when I have something to show.

**Comment by eLindros** [comment]
Hi,

I have implemented this feature in a separate node-express-app. You can find and use it from here: [kickbase-autodecline](https://github.com/eLindros/kickbase-autodecline). 

One has to provide the Kickbase credentials as environment variables on the server. The schedule is configurable.

I've used it now for a couple of days and it reduced my stress level considerable :-)

Use it on your own risk ;-) and let me know what you think.
